src/hotspot/share/utilities/debug.cpp

```diff
@@ -361,10 +361,14 @@ void report_java_out_of_memory(const char* message) {
 
     if (CrashOnOutOfMemoryError) {
       tty->print_cr("Aborting due to java.lang.OutOfMemoryError: %s", message);
+      // SapMachine 2021-05-21: Print stack to stdout.
+      VMError::print_stack(tty);
```

Comment:
I'm not sure if it is the best idea to do a heap dump for thread exhaustion too, since it can take a lot of time. And if the VM is killed during dumping of the heap we are missing the thread dump, which is probably more useful in this case, Plus it would be nice to have the thread dump even if we are not exiting the VM.

Reply:
Just to see if I understand, you propose:
- HeapDumpOnOutOfMemoryError should only be active for OOMs from Heap and Metaspace, whereas the other three switches should work for thread exhaustion too
- A stack should be written if any of these four switches is active? So, if -XX:OnOutOfMemoryError=myscript, we should print the stack too?